hooks/useUpdateTokenMetadata.ts

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react'
+import axios, { AxiosError } from 'axios'
+import { useGetPoint } from '@/hooks/quest/useGetPoint'
+
+export interface TokenAttribute {
+  display_type: string | undefined
+  trait_type: string
+  value: number | string
+}
+export interface KamonToken {
+  name: string
+  description: string
+  image: string
+  attributes: TokenAttribute[]
+}
+
+export const useUpdateTokenMetadata = () => {  
+  const { refetchPoint } = useGetPoint()
+  const [updateTokenMetadataIsSubmitting, setUpdateTokenMetadataIsSubmitting] = useState<boolean>()
+  const ipfsApiEndpoint = process.env.NEXT_PUBLIC_IPFS_API_URI + ''
+
+  const updateTokenMetadata = async (tokenJSON: KamonToken, userAddress: string) => {
+    setUpdateTokenMetadataIsSubmitting(true)
+    let dateFromToken = 0
+    let rolesFromToken: string[] = []
+    const updatedPointsQuery = await refetchPoint()
+    const updatedPoints = Array.isArray(updatedPointsQuery.data)? updatedPointsQuery.data[0]: 0
+    const updatedPointsInt = parseInt(updatedPoints.toString())
+    tokenJSON?.attributes.forEach((attr: TokenAttribute) => {
+      if (attr.trait_type == 'Date') {
+        dateFromToken = parseInt(attr.value.toString())
+      } else if (attr.trait_type == 'Role') {
+        rolesFromToken.push(attr.value.toString())
+      }
+    })
+    const payload = {
+      address: userAddress,
+      roles: rolesFromToken,
+      points: updatedPointsInt,
+      date: dateFromToken,
+    }
+    try {
+      const ipfsRequest = await axios.post(ipfsApiEndpoint, payload, {
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      })
+      setUpdateTokenMetadataIsSubmitting(false)
+      return(ipfsRequest.data.tokenUri)
+    } catch (err) {
```

Comment:
I think we don't need to catch the error.
if it makes an error, just make it happen.
so that the user will notice oh something wrong.
not always catching errors is good.
errors occur because something is wrong, so you don't need to hide.

and if you remove the try {} statement, this file will be slimmer too

Reply:
Done in https://github.com/henkaku-center/omise-interface/pull/80/commits/f76e4dd4ee249241bd6da616810c586779d71bbf